src/compiler/factory/nodeFactory.ts

```diff
@@ -6553,6 +6553,8 @@ export function createNodeFactory(flags: NodeFactoryFlags, baseFactory: BaseNode
                 return updateSatisfiesExpression(outerExpression, expression, outerExpression.type);
             case SyntaxKind.NonNullExpression:
                 return updateNonNullExpression(outerExpression, expression);
+            case SyntaxKind.ExpressionWithTypeArguments:
+                return updateExpressionWithTypeArguments(outerExpression, expression, outerExpression.typeArguments);

```

Comment:
I will note that most of the cases here are uncovered by tests: 
![image](https://github.com/user-attachments/assets/54c23c2d-0db7-4de6-a757-dda8cc525ee6)